iiif_prezi3/helpers/set_hwd_from_file.py

```diff
@@ -0,0 +1,37 @@
+import mimetypes
+
+from PIL import Image
+
+from ..loader import monkeypatch_schema
+from ..skeleton import Canvas
+
+
+class SetHeightWidthDurationFileHelper:
+    """Introspect a file and set the height, width, and duration properties.
+
+    Args:
+        file_path_or_object (filepath or fileobject): the file path or file object to introspect
+
+    Returns:
+        None
+    """
+
+    def set_hwd_from_file(self, file_path_or_object):
+        if isinstance(file_path_or_object, str):
+            filetype, _ = mimetypes.guess_type(file_path_or_object)
+            if not filetype.startswith("image/"):
+                raise NotImplementedError
+
```

Comment:
Can you add an elif to handle being passed a file object? I think the check would be `isinstance(file_path_or_object, io.BufferedIOBase)` which should cover all binary file-like objects (if someone is silly enough to open an image without the `b` mode flag, that's their lookout!)

The filename should be accessible via the `.name` property,  but it might not be if passed a raw bytestream of the file, in which case we should probably just raise an error.

Reply:
And maybe an else clause that throws TypeErrror (in case anyone tries to pass an int or something that will make PIL error out anyway)